File: LEVEL3/순위.cpp
#include<bits/stdc++.h>
using namespace std;

int solution(int n, vector<vector<int>> results) {
    int answer = 0;
    vector<vector<bool>> chk(n + 1, vector<bool>(n + 1, false));
    
    for(auto i : results) {
        int win = i[0];
        int lose = i[1];
        chk[win][lose] = true;
    }
    
    for(int via = 1; via <= n; via++) {
        for(int from = 1; from <= n; from++) {
            for(int to = 1; to <= n; to++) {
                if(chk[from][via] && chk[via][to]) {
                    chk[from][to] = true;
                }
            }
        }
    }
    
    for(int i = 1; i <= n; i++) {
        int cnt = 0;
        
        for(int j = 1; j <= n; j++) {
            if(i == j) {
                continue;
            }
            
            if(chk[i][j] || chk[j][i]) {
                cnt++;
            }
        }
        
        if(cnt == n - 1) {
            answer++;
        }
    }
    
    return answer;
}

File: LEVEL3/순위.java
class Solution {
    public int solution(int n, int[][] results) {
        int answer = 0;
        boolean[][] chk = new boolean[n + 1][n + 1];
        
        for(int i = 0; i < results.length; i++) {
            chk[results[i][0] - 1][results[i][1] - 1] = true;
        }
        
        for(int i = 0; i < n; i++) {
            for(int j = 0; j < n; j++) {
                for(int k = 0; k < n; k++) {
                    if(chk[j][i] && chk[i][k]) {
                        chk[j][k] = true;
                    }
                }
            }
        }
        
        for(int i = 0; i < n; i++) {
            int cnt = 0;
            
            for(int j = 0; j < n; j++) {
                if(i == j) {
                    continue;
                }
                
                if(chk[i][j] || chk[j][i]) {
                    cnt++;
                }
            }
            
            if(cnt == n - 1) {
                answer++;
            }
        }
        
        return answer;
    }
}